spanner/spanner_snippets/snippet.go

```diff
@@ -1266,6 +1277,329 @@ func updateUsingBatchDML(ctx context.Context, w io.Writer, client *spanner.Clien
 
 // [END spanner_dml_batch_update]
 
+// [START spanner_create_table_with_datatypes]
+
+func createTableWithDatatypes(ctx context.Context, w io.Writer, adminClient *database.DatabaseAdminClient, database string) error {
+	op, err := adminClient.UpdateDatabaseDdl(ctx, &adminpb.UpdateDatabaseDdlRequest{
+		Database: database,
+		Statements: []string{
+			`CREATE TABLE Venues (
+				VenueId	INT64 NOT NULL,
+				VenueName STRING(100),
+				VenueInfo BYTES(MAX),
+				Capacity INT64,
+				AvailableDates ARRAY<DATE>,
+				LastContactDate DATE,
+				OutdoorVenue BOOL,
+				PopularityScore FLOAT64,
+				LastUpdateTime TIMESTAMP NOT NULL OPTIONS (allow_commit_timestamp=true)
+			) PRIMARY KEY (VenueId)`,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	if err := op.Wait(ctx); err != nil {
+		return err
+	}
+	fmt.Fprintf(w, "Created Venues table in database [%s]\n", database)
+	return nil
+}
+
+// [END spanner_create_table_with_datatypes]
+
+// [START spanner_insert_datatypes_data]
+
+func writeDatatypesData(ctx context.Context, w io.Writer, client *spanner.Client) error {
+	venueColumns := []string{"VenueId", "VenueName", "VenueInfo", "Capacity", "AvailableDates",
+		"LastContactDate", "OutdoorVenue", "PopularityScore", "LastUpdateTime"}
+	exampleBytes1 := []byte("Hello World 1")
+	exampleBytes2 := []byte("Hello World 2")
+	exampleBytes3 := []byte("Hello World 3")
+	var availableDates1 = []string{"2020-12-01", "2020-12-02", "2020-12-03"}
+	var availableDates2 = []string{"2020-11-01", "2020-11-05", "2020-11-15"}
+	var availableDates3 = []string{"2020-10-01", "2020-10-07"}
+	m := []*spanner.Mutation{
+		spanner.InsertOrUpdate("Venues", venueColumns,
+			[]interface{}{4, "Venue 4", exampleBytes1, 1800, availableDates1,
+				"2018-09-02", false, 0.85543, spanner.CommitTimestamp}),
+		spanner.InsertOrUpdate("Venues", venueColumns,
+			[]interface{}{19, "Venue 19", exampleBytes2, 6300, availableDates2,
+				"2019-01-15", true, 0.98716, spanner.CommitTimestamp}),
+		spanner.InsertOrUpdate("Venues", venueColumns,
+			[]interface{}{42, "Venue 42", exampleBytes3, 3000, availableDates3,
+				"2018-10-01", false, 0.72598, spanner.CommitTimestamp}),
+	}
+	_, err := client.Apply(ctx, m)
+	return err
+}
+
+// [END spanner_insert_datatypes_data]
+
+// [START spanner_query_with_array_parameter]
+
+func queryWithArray(ctx context.Context, w io.Writer, client *spanner.Client) error {
+	var date1 = civil.Date{Year: 2020, Month: time.October, Day: 1}
+	var date2 = civil.Date{Year: 2020, Month: time.November, Day: 1}
+	var exampleArray = []civil.Date{date1, date2}
+	stmt := spanner.Statement{
+		SQL: `SELECT VenueId, VenueName, AvailableDate FROM Venues v,
+            UNNEST(v.AvailableDates) as AvailableDate 
+            WHERE AvailableDate IN UNNEST(@availableDates)`,
```

Comment:
Indenting looks a little odd - maybe increase the indent a bit?

Reply:
Updated.